### ISSUE TITLE
Graphs crash with latest (6.2.0) release

### ISSUE THREAD

**Issue by jacobwilliams** [Fortran-FOSS-Programmers/ford#487]
I'm getting an "Unrecognised object type 'str' when constructing graphs" crash for [this code](https://github.com/jacobwilliams/dvode/tree/9426c8f1d02030e313bb69d796b5836aff5eabc6), which didn't have any problems with the 6.1.17 release.

Traceback is:

```
Traceback (most recent call last):
  File "/miniconda3/bin/ford", line 8, in <module>
    sys.exit(run())
  File "/python3.9/site-packages/ford/__init__.py", line 646, in run
    main(proj_data, proj_docs, md)
  File "/python3.9/site-packages/ford/__init__.py", line 630, in main
    docs = ford.output.Documentation(proj_data, proj_docs_, project, page_tree)
  File "/python3.9/site-packages/ford/output.py", line 158, in __init__
    self.graphs.register(item)
  File "/python3.9/site-packages/ford/graphs.py", line 1255, in register
    self.data.register(obj)
  File "/python3.9/site-packages/ford/graphs.py", line 194, in register
    collection[obj] = NodeType(obj, self, hist)
  File "/python3.9/site-packages/ford/graphs.py", line 311, in __init__
    n = gd.get_node(u)
  File "/python3.9/site-packages/ford/graphs.py", line 212, in get_node
    collection, _ = self._get_collection_and_node_type(obj)
  File "/python3.9/site-packages/ford/graphs.py", line 172, in _get_collection_and_node_type
    raise BadType(
ford.graphs.BadType: "Unrecognised object type 'str' when constructing graphs"
```

**Comment by ZedThree** [comment]
Thanks for the bug report @jacobwilliams, it let me find a few other bugs at the same time! It'll probably be tomorrow at least before I get a new version out I'm afraid

**Comment by ZedThree** [comment]
This is fixed in 6.2.1
